fund_store/scripts/data_updates/FLS-907-update-la-names.py

```diff
@@ -0,0 +1,525 @@
+
+        assessment_records = db.session.scalars(
+            select(AssessmentRecord).where(AssessmentRecord.fund_id.in_([round.fund_id]))
+        ).all()
+        print(f"\n\n{fund.short_name} AssessmentRecord count: {len(assessment_records)}\n")
+
+        for assessment_record in assessment_records:
+            project_name = assessment_record.project_name
+            json_project_name = assessment_record.jsonb_blob["project_name"]
+
+            print(f"\nAssessmentRecord: {assessment_record.application_id}, {project_name}")
+            if project_name in CODE_TO_NAME_MAP:
+                print(
+                    f"AssessmentRecord.project_name: {project_name} ",
+                    f"replacing with: {CODE_TO_NAME_MAP[project_name]}",
+                )
+                project_name = CODE_TO_NAME_MAP[project_name]
+            else:
+                print(f"AssessmentRecord.project_name not found: {project_name}")
+
+            new_json_2 = copy.deepcopy(assessment_record.jsonb_blob)
+
+            print(
+                f"AssessmentRecord.json_project_name: {assessment_record.application_id},",
+                f"{json_project_name}",
+            )
+            if json_project_name in CODE_TO_NAME_MAP:
+                print(
+                    f"AssessmentRecord.json_project_name: {json_project_name} ",
+                    f"replacing with: {CODE_TO_NAME_MAP[json_project_name]}",
+                )
+                new_json_2["project_name"] = CODE_TO_NAME_MAP[json_project_name]
+            else:
+                print(f"AssessmentRecord.json_project_name not found: {json_project_name}")
+
+            for form in new_json_2["forms"]:
+                if form["name"] not in [
+                    "local-plans-local-authority-details-signed-off",
+                    "local-plans-local-authority-details",
+                ]:
+                    continue
+
+                for question in form["questions"]:
+                    for field in question["fields"]:
+                        if field["key"] not in ["RoLhhf", "OlCBjB"]:
+                            continue
+
+                        if field["answer"] in CODE_TO_NAME_MAP:
+                            print(
+                                f"Form.answer: {field['answer']} ->",
+                                f"replacing with: {CODE_TO_NAME_MAP[field['answer']]}",
+                            )
+                            field["answer"] = CODE_TO_NAME_MAP[field["answer"]]
+                        else:
+                            print(f"From answer not found: {field['key']}, {field['answer']}")
+
+            # update the json
+            assessment_record.jsonb_blob = new_json_2
+
+    if commit:
+        db.session.commit()
+
+
+def main():
+    update_local_authority_name(commit=False)
+
+    # ask for confirmation
+    if input("Commit changes? (y/n): ").lower() == "y":
+        update_local_authority_name(commit=True)
```

Comment:
I assume the db session changes from the first run of the `update_local_authority_name` method will be discarded or overwritten? a bit strange to run it twice but I can't think of anything it would do wrong

Reply:
It was a simple way to get the `print()` to be echo'd 1st, so I can inspect what changes would be made. Then only on confirm have them executed. In hind-sight, it could perhaps just execute the commit, but re-printing everything is no the worst.